lib/ui/painting/codec.cc

```diff
@@ -362,13 +363,29 @@ MultiFrameCodec::MultiFrameCodec(std::unique_ptr<SkCodec> codec)
     : codec_(std::move(codec)) {
   repetitionCount_ = codec_->getRepetitionCount();
   frameInfos_ = codec_->getFrameInfo();
-  frameBitmaps_.resize(frameInfos_.size());
+  frameBitmaps_.clear();
+  // Initialize the frame cache, marking frames that are required for other
+  // dependent frames to render.
+  for (size_t frameIndex = 0; frameIndex < frameInfos_.size(); frameIndex++) {
+    const auto& frameInfo = frameInfos_[frameIndex];
+    if (frameInfo.fRequiredFrame != SkCodec::kNoFrame) {
+      frameBitmaps_[frameInfo.fRequiredFrame] =
+          std::make_unique<DecodedFrame>(SkBitmap(), /*required=*/true);
+    }
+    if (cacheAllFrames_ && frameBitmaps_.count(frameIndex) < 1) {
+      frameBitmaps_[frameIndex] =
+          std::make_unique<DecodedFrame>(SkBitmap(), /*required=*/false);
+    }
+  }
   nextFrameIndex_ = 0;
 }
 
 sk_sp<SkImage> MultiFrameCodec::GetNextFrameImage(
     fml::WeakPtr<GrContext> resourceContext) {
-  SkBitmap& bitmap = frameBitmaps_[nextFrameIndex_];
+  SkBitmap newBitmap;
+  SkBitmap& bitmap = cacheAllFrames_ && frameBitmaps_.count(nextFrameIndex_) > 0
```

Comment:
Don't we want to use a cached frame even if cacheAllFrames_ is false?

Reply:
Yes, fixed.